### ISSUE TITLE
🐛 Error when trying to set up computer with private SSH key

### ISSUE THREAD

**Issue by mbercx** [aiidalab/aiidalab-widgets-base#261]
When trying to set up the SSH connection to Eiger using a private key, I run into the following error:

```Python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~/.local/lib/python3.7/site-packages/aiidalab_widgets_base/computers.py in on_setup_ssh(self, change)
    395         with self._setup_ssh_out:
    396             mode = self._verification_mode.value
--> 397             self._on_setup_ssh(mode, change)
    398 
    399     def _on_setup_ssh(

~/.local/lib/python3.7/site-packages/aiidalab_widgets_base/computers.py in _on_setup_ssh(self, mode, change)
    451 
    452         # get the right commands to access the proxy server (if provided)
--> 453         success, proxycmd = self._configure_proxy(password, proxy_password)
    454         if not success:
    455             return

~/.local/lib/python3.7/site-packages/aiidalab_widgets_base/computers.py in _configure_proxy(self, password, proxy_password)
    380 
    381             # Finally trying to connect.
--> 382             if self._send_pubkey(self.proxy_hostname, proxy_username, proxy_password):
    383                 return True, proxy_username + "@" + self.proxy_hostname
    384 

~/.local/lib/python3.7/site-packages/aiidalab_widgets_base/computers.py in _send_pubkey(hostname, username, password, proxycmd)
    324         }
    325         if index == 0:
--> 326             child.sendline(password)
    327             try:
    328                 child.expect("Now try logging into", timeout=5)

/opt/conda/lib/python3.7/site-packages/pexpect/pty_spawn.py in sendline(self, s)
    575         line in the default terminal mode, see docstring of :meth:`send`.
    576         '''
--> 577         s = self._coerce_send_string(s)
    578         return self.send(s + self.linesep)
    579 

/opt/conda/lib/python3.7/site-packages/pexpect/spawnbase.py in _coerce_send_string(self, s)
    144     def _coerce_send_string(self, s):
    145         if self.encoding is None and not isinstance(s, bytes):
--> 146             return s.encode('utf-8')
    147         return s
    148 

AttributeError: 'NoneType' object has no attribute 'encode'
```

It seems the current code expects the "password" field to be filled in? If I try to set up the SSH connection with the password, it seems to work fine.

**Comment by csadorf** [comment]
@yakutovicha Can you have a look at this?

**Comment by yakutovicha** [comment]
yes

**Comment by yakutovicha** [comment]
fixed in #279
